### ISSUE TITLE
Using `gcloud preview app run` for Managed VMs results in no running docker containers

### ISSUE THREAD

**Issue by m4tty** [GoogleCloudPlatform/gcloud#33]
Is this expected (when using managed vms... i.e. configured with vm:true)? 

Because gcloud is using dev_appserver.py (instead of the docker tool chain?) I think I've run into a number of related issues (e.g. appenginevm build tag isn't set which causes trouble with golang/oauth2 - See https://github.com/golang/oauth2/issues/142. )  

I'm also doing some work in Java Managed VMs and running `mvn gcloud:run` has different results (e.g. docker containers running, and logs / stdout redirected to the docker container) than running `gcloud preview app run` after creating a yaml file via appcfg.sh (which results in no running containers).

Am I misunderstanding how `gcloud preview app run` is supposed to function?  It does appear to depend on docker (it outputs INFO: Looking for the Dockerfile and Using Dockerfile) but ultimately no containers are running and the app is launched via dev_appserver.py.

Thanks!

**Comment by m4tty** [comment]
Ok, based on this thread (https://code.google.com/p/google-cloud-sdk/issues/detail?id=163) it seems that defaulting to docker-less development now happens via gcloud.  And when in docker-less dev mode the `appenginevm` build flag is not passed to the go build by gcloud (which has some negative effects (on oauth2 at least)).  When docker is used (go-wrapper is run which sets the build flag) things work as expected.  It seems that `mvn gcloud:run` defaults to using docker-ized dev mode.  Using `mvn gcloud:run` is the only way I've been able to run in a dockerized dev environment.  (Is there a way to turn on dockerized dev using `gcloud preview app run`?)

Long story short, there appears to be an issue with `gcloud preview app run` or it's tooling (when in docker-less dev) that it doesn't pass the necessary build flags to the go build.

Note: I'm also investigating a logging issue when in docker-less dev via gcloud in java.  It appears that logs do not make it to the primary console window.  Logging works as expected in a dockerized deploy (via mvn gcloud:run), as logs can be found in the docker container.
